src/plugins/kconfig/README.md

```diff
@@ -0,0 +1,98 @@
+- infos = Information about the kconfig plugin is in keys below
+- infos/author = Dardan Haxhimustafa <mail@dardan.im>
+- infos/licence = BSD
+- infos/needs =
+- infos/provides = storage/kconfig
+- infos/recommends =
+- infos/placements = getstorage setstorage
+- infos/status = recommended maintained compatible specific experimental unfinished nodoc concept
+- infos/metadata =
+- infos/description = Reads the KConfig INI format
+
+## Introduction
+
+This plugin can be used to parse a [KConfig](https://cgit.kde.org/kconfig.git) INI file into a KeySet.
+
+Information about the syntax:
+
+- Files are expected to be encoded in `UTF-8`.
+- Empty lines are ignored.
+- Lines that start with a `#` character are considered comments. Comments are ignored too.
+- Configurations consist of groups and keys. Only keys can have values.
+- Key names can't start with a `[` character
+- Keys can contain spaces and any special characters as long except `=`.
+- If a key has a value, then it will be followed with an `=` symbol and then the value will be read until the end of the line.
+  The white space characters around the `=` symbol are ignored.
+- In values, the following escape sequences can be used:
+  - `\n` and `r` are mapped to newline
+  - `\t` is mapped to tab
+  - `\\` is mapped to `\`
+- Values can contain any character from the `UTF-8` set except for newline and `\` followed by an invalid excape sequence.
+- Keys can be localized. The locale is surrounded with `[` and `]` and cannot contain start with `$`.
+- Same key name can be used multiple times if it has different locales. The following example is valid:
+  ```
+  greeting[en] = Hello
+  greeting[de] = Hallo
+  ```
+- Keys can have meta data. Those are one character (1 byte only) long, start with `$` and are surrounded with `[` and `]`.
+- Same key name can't be used multiple times with different metadata as in locales. The following example is invalid:
+  ```
+  key.name[$a] = Something
+  key.name[$i] = Something else
+  ```
+- If a key `keyname[$metavalue]` is parsed, it will be represented as a Key with name `parent/keyname` and meta `kconfig` as `metavalue`
```

Comment:
I thought only a single character or byte is allowed?

Reply:
yes. I just used `metavalue` as a placeholder for a single-byte value, which isn't a smart idea. I'll change it to `a`

Reply:
Or should I remove this list item at all since I show how it's represented in an example below?

Reply:
Yes, you can remove it.